### DISCUSSION TITLE
How to solve the N+1 problem ?

### DISCUSSION THREAD

**Discussion by malisetti** [MichalLytek/type-graphql#909] [Q&A]
Are there any solutions to the N+1 problem ? I am using type-graphql with typeorm.

**Comment by lauriharpf** [comment]
Have you experimented with https://github.com/slaypni/type-graphql-dataloader? It is intended to solve this problem.

There's also a related open type-graphql issue https://github.com/MichalLytek/type-graphql/issues/51, browsing through it might be helpful. Plenty of good comments there, I found https://github.com/MichalLytek/type-graphql/issues/51#issuecomment-670326992 and https://github.com/MichalLytek/type-graphql/issues/51#issuecomment-626719854 especially helpful.

I'm also experiencing this issue, but using type-graphql with RESTDataSources instead of typeorm. Haven't had time to try to implement those suggested solutions yet.
